### ISSUE TITLE
dimensionality mismatch: Decision Tree CLI with both -t and -T specified

### ISSUE THREAD

**Issue by ydu20** [mlpack/mlpack#3719]
#### Issue description

I tried running the decision tree cli command with the iris dataset, but the program runs into an error regarding the dimension of the test set during the prediction phase.

I believe the issue is in line 195~200 of decision_tree_main.cpp, where in the case when trainingSet contains the labels, its DatasetInfo object is not updated to reflect the reduced dimensionality. As a result, at line 271-272 the loading of the test dataset results in a dimension mismatch.

#### Your environment

 * version of mlpack: part of mlpack git-f12ec422dc.
 * operating system: Ubuntu 22.04.4
 * compiler: CLI
 * version of dependencies (Boost/Armadillo): Boost -> 1.74.0.3ubuntu7, Armadillo -> 1:10.8.2+dfsg-1
 * any other environment information you think is relevant:

#### Steps to reproduce

mlpack_decision_tree -t datasets/iris_train.csv -T datasets/iris_test.csv -v -a -p mlpack_dct_predict.csv

#### Actual behavior

[INFO ] Loading 'datasets/iris_train.csv' as CSV dataset.  Size is 150 x 5.
[INFO ] Using the last dimension of training set as labels.
[INFO ] 96% correct on training set (144 / 150).
[INFO ] Loading 'datasets/iris_test.csv' as CSV dataset.  [FATAL] data::LoadCSV(): given DatasetInfo has dimensionality 5, but data has dimensionality 4

terminate called after throwing an instance of 'std::runtime_error'
  what():  fatal error; see Log::Fatal output
Aborted (core dumped)
